apps/app_rpt.c

```diff
@@ -1335,101 +1336,112 @@ void *rpt_call(void *this)
 			ast_free(instr);
 		}
 	}
-
 	ast_channel_context_set(mychannel, myrpt->patchcontext);
 	ast_channel_exten_set(mychannel, myrpt->exten);
 
 	if (myrpt->p.acctcode)
 		ast_channel_accountcode_set(mychannel, myrpt->p.acctcode);
 	ast_channel_priority_set(mychannel, 1);
 	ast_channel_undefer_dtmf(mychannel);
-	if (ast_pbx_start(mychannel) < 0) {
-		ast_log(LOG_ERROR, "Unable to start PBX!\n");
+
+	if (rpt_call_bridge_setup(myrpt, mychannel)) {
+		rpt_mutex_lock(&myrpt->lock);
+		myrpt->callmode = CALLMODE_DOWN;
+		rpt_mutex_unlock(&myrpt->lock);
 		ast_hangup(mychannel);
 		ast_hangup(genchannel);
+		pthread_exit(NULL);
+	}
+
+	if (ast_pbx_start(mychannel) != AST_PBX_SUCCESS) {
+		ast_log(LOG_ERROR, "Unable to start PBX!\n");
 		rpt_mutex_lock(&myrpt->lock);
 		myrpt->callmode = CALLMODE_DOWN;
 		rpt_mutex_unlock(&myrpt->lock);
+		ast_hangup(mychannel);
+		ast_hangup(genchannel);
 		pthread_exit(NULL);
 	}
-	usleep(10000);
 	rpt_mutex_lock(&myrpt->lock);
 	myrpt->callmode = CALLMODE_UP;
-
-	if (ast_channel_pbx(mychannel)) {
-		if (rpt_call_bridge_setup(myrpt, mychannel, genchannel)) {
-			myrpt->callmode = CALLMODE_DOWN;
-			rpt_mutex_unlock(&myrpt->lock);
-			pthread_exit(NULL);
-		}
-	} else {
-		/* XXX Can this ever happen (since we exit if ast_pbx_start fails)? */
-		ast_log(LOG_WARNING, "%s has no PBX?\n", ast_channel_name(mychannel));
-	}
-
 	sentpatchconnect = 0;
 	while (myrpt->callmode != CALLMODE_DOWN) {
-		if (!ast_channel_pbx(mychannel) && (myrpt->callmode != CALLMODE_FAILED)) {
+		channel_ref = ast_channel_get_by_name(mychannel_name);
+		if (!channel_ref && (myrpt->callmode != CALLMODE_FAILED)) {
 			/* If patch is setup for far end disconnect */
 			if (myrpt->patchfarenddisconnect || (myrpt->p.duplex < 2)) {
-				ast_debug(1, "callmode=%i, patchfarenddisconnect=%i, duplex=%i\n",
-					myrpt->callmode, myrpt->patchfarenddisconnect, myrpt->p.duplex);
+				ast_debug(1,
+					"callmode=%i, patchfarenddisconnect=%i, duplex=%i\n",
+					myrpt->callmode,
+					myrpt->patchfarenddisconnect,
+					myrpt->p.duplex);
```

Comment:
This is still not what @InterLinked1 wants to see.
We need a callback similar to the `statis` version, just not stasis?
I haven't been able to locate a method for creating a callback for a channel. If you have one handy it would be much appreciated.